templates/single-product/add-to-cart/grouped.php

```diff
@@ -24,64 +25,73 @@
+				'label',
+				'price',
+			), $product );
+
+			foreach ( $grouped_products as $grouped_product ) {
+				$post_object        = get_post( $grouped_product->get_id() );
+				$quantites_required = $quantites_required || ( $grouped_product->is_purchasable() && ! $grouped_product->has_options() );
+
+				setup_postdata( $post = $post_object ); // WPCS: override ok.
+
+				echo '<tr id="product-' . esc_attr( get_the_ID() ) . '" class="woocommerce-grouped-product-list-item ' . esc_attr( implode( ' ', get_post_class() ) ) . '">';
+
+				// Output columns for each product.
+				foreach ( $grouped_product_columns as $column_id ) {
+					do_action( 'woocommerce_grouped_product_list_before_' . $column_id, $grouped_product );
+
+					switch ( $column_id ) {
+						case 'quantity' :
+							ob_start();
+
+							if ( ! $grouped_product->is_purchasable() || $grouped_product->has_options() ) {
+								woocommerce_template_loop_add_to_cart();
+							} elseif ( $grouped_product->is_sold_individually() ) {
+								echo '<input type="checkbox" name="' . esc_attr( 'quantity[' . $grouped_product->get_id() . ']' ) . '" value="1" class="wc-grouped-product-add-to-cart-checkbox" />';
+							} else {
+								do_action( 'woocommerce_before_add_to_cart_quantity' );
+
+								woocommerce_quantity_input( array(
+									'input_name'  => 'quantity[' . $grouped_product->get_id() . ']',
+									'input_value' => isset( $_POST['quantity'][ $grouped_product->get_id() ] ) ? wc_stock_amount( wc_clean( wp_unslash( $_POST['quantity'][ $grouped_product->get_id() ] ) ) ) : 0, // WPCS: CSRF ok, input var okay, sanitization ok.
+									'min_value'   => apply_filters( 'woocommerce_quantity_input_min', 0, $grouped_product ),
+									'max_value'   => apply_filters( 'woocommerce_quantity_input_max', $grouped_product->get_max_purchase_quantity(), $grouped_product ),
+								) );
+
+								do_action( 'woocommerce_after_add_to_cart_quantity' );
+							}
+
+							$value = ob_get_clean();
+							break;
+						case 'label' :
+							$value  = '<label for="product-' . esc_attr( $grouped_product->get_id() ) . '">';
+							$value .= $grouped_product->is_visible() ? '<a href="' . esc_url( apply_filters( 'woocommerce_grouped_product_list_link', get_permalink( $grouped_product->get_id() ), $grouped_product->get_id() ) ) . '">' . $grouped_product->get_name() . '</a>' : $grouped_product->get_name();
+							$value .= '</label>';
+							break;
+						case 'price' :
+							$value = $grouped_product->get_price_html() . wc_get_stock_html( $grouped_product );
+							break;
+						default :
+							$value = '';
+							break;
+					}
+
+					echo '<td class="woocommerce-grouped-product-list-item__' . esc_attr( $column_id ) . ' ' . esc_attr( $column_id ) . '">' . apply_filters( 'woocommerce_grouped_product_list_column_' . $column_id, $value, $grouped_product ) . '</td>'; // WPCS: XSS ok.
+
+					do_action( 'woocommerce_grouped_product_list_after_' . $column_id, $grouped_product );
```

Comment:
@mikejolley the before and after hooks are *outside* the `<td>` element. IMO, they aren't that useful that way because the content added to that hook displays outside the table entirely.  ie: I'd expect something on the `before_price` hook to display near the price, and not above the table. I though I see I can use the filter to add content to the cell, I'm curious why add the hooks outside the cell?

Reply:
@helgatheviking They were there before your patch - moving them like you did breaks BW compat. I added a matching 'after', and a filter to the contents. The filter is all you need to prepend/append/change content now.

Reply:
woocommerce_grouped_product_list_column_X

Reply:
@mikejolley Is that really breaking BW compat or is it fixing a bug? :) b/c having the hooks outside the `<td>` element seems pointless to me. Adding the *after* hook intentionally in a useless place seems like doubling down on a past mistake. Am I missing something about the utility of the hooks that display outside of the table?

Reply:
You could add a new TD via it?

Reply:
AH ha! Ok, that makes more sense to me now. Still some hooks inside would be cool, but yes I can get Name Your Price support in via the filter now.